packages/core-flows/src/customer/steps/update-addresses.ts

```diff
@@ -0,0 +1,54 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import {
+  CustomerAddressDTO,
+  FilterableCustomerAddressProps,
+  ICustomerModuleService,
+} from "@medusajs/types"
+import {
+  getSelectsAndRelationsFromObjectArray,
+  promiseAll,
+} from "@medusajs/utils"
+import { createStep, StepResponse } from "@medusajs/workflows-sdk"
+
+type UpdateCustomerAddresseStepInput = {
+  selector: FilterableCustomerAddressProps
+  update: Partial<CustomerAddressDTO>
+}
+
+export const updateCustomerAddresseStepId = "update-customer-addresses"
+export const updateCustomerAddressesStep = createStep(
+  updateCustomerAddresseStepId,
+  async (data: UpdateCustomerAddresseStepInput, { container }) => {
+    const service = container.resolve<ICustomerModuleService>(
+      ModuleRegistrationName.CUSTOMER
+    )
+
+    const { selects, relations } = getSelectsAndRelationsFromObjectArray([
+      data.update,
+    ])
+    const prevCustomers = await service.listAddresses(data.selector, {
+      select: selects,
+      relations,
+    })
+
+    const customerAddresses = await service.updateAddress(
+      data.selector,
+      data.update
+    )
+
+    return new StepResponse(customerAddresses, prevCustomers)
+  },
+  async (prevCustomerAddresses, { container }) => {
+    if (!prevCustomerAddresses) {
+      return
+    }
+
+    const service = container.resolve<ICustomerModuleService>(
+      ModuleRegistrationName.CUSTOMER
+    )
+
+    await promiseAll(
+      prevCustomerAddresses.map((c) => service.updateAddress(c.id, { ...c }))
```

Comment:
is there a way to update that in bulk?

Reply:
Not right now - we could add an `upsertAddress`, but would like the `update*` signatures to be `(selector, updateData)` instead of `({ id, ...updateData })`

@adrien2p - is `upsert` baked into the abstract services?

Reply:
it is 💃🏻